### ISSUE TITLE
Update to Apache 

### ISSUE THREAD

**Issue by elharo** [googleapis/google-http-java-client#1205]
HttpClient 5.0
HttpCore 5.1 

This will be required to support  HTTP/2. This is a big one so needs to be investigated carefully.

**Comment by chingor13** [comment]
We likely would want another separate artifact like `google-http-client-apache5`

**Comment by chingor13** [comment]
We also probably need to drop the original apache client (that doesn't really work) that is packaged in `google-http-client`.

**Comment by elharo** [comment]
I don't think we need a new google-http-client-apache5 and we can simply drop the old one. That is, this is no longer used. The real update needs to happen in core.

**Comment by elharo** [comment]
On further investigation, we probably do need a new artifact for this because the existing API exposes org.apache.http.client types in its public API and those types have changed in HTTP client 5. We probably shouldn't have done that, but that ship sailed long ago. 

This time around let's see if we can avoid exposing third party types  on our API surface.

**Comment by chingor13** [comment]
For context, we accept a fully configured Apache HttpClient instance otherwise we would basically be reimplementing the entire surface of the HttpClient builder to provide users the ability to configure the underlying Apache transport.

**Comment by meor9zcu12** [comment]
HTTP/2 is very important in my case, is there any plan to upgrade to HttpClient 5?
Some libraries use `google-http-java-client` as dependency, and hence I can't easily customize and override the Apache HttpClient  implementation in these libraries.

**Comment by bc-bartvb** [comment]
So, what is the status. if any? Any plans? Our migration to Apache HTTP 5.x is now stuck because of this. If there are no plans, we will move away from using this library. There are now dependencies that are EOL.

**Comment by mjschwaiger** [comment]
Are there any updates on this topic?
Last release of [httpclient (version 4)](https://mvnrepository.com/artifact/org.apache.httpcomponents/httpclient) was in November 2022 - more than 2 years ago ...

Please keep us informed.
